packages/integration-sdk-runtime/src/execution/step.ts

```diff
@@ -163,12 +169,27 @@ export function prepareLocalStepCollection<
           ctx,
         ) ?? {});
         const enabledRecord: StepStartStates = {};
+
+        if (dependenciesCache?.enabled) {
+          ctx.logger.info(
+            `Using dependencies cache found at ${dependenciesCache.filepath}`,
+          );
+        }
+
         for (const stepId of allStepIds) {
           const originalValue = originalEnabledRecord[stepId] ?? {};
           if (stepsToRun.includes(stepId)) {
             enabledRecord[stepId] = {
               ...originalValue,
               disabled: false,
+              ...(dependenciesCache?.enabled &&
```

Comment:
Would you mind extracting this out into a function? I think it will be a bit easier for readers to understand.